### PR TITLE
test(yaml): add `parse()` test for duplicate keys

### PR DESCRIPTION
**Pull request by iuioiua**
Unblocks #5240

### CODE REVIEW

File: yaml/parse_test.ts
Comment (iuioiua) [inline yaml/parse_test.ts:1]: Current documentation says that if `ParseOptions.json` is `true`, then duplicate keys cause `parse()` to throw. By default, `ParseOptions.json` is `false`, and duplicate keys cause `parse()` to throw. In other words, `ParseOptions.json` currently does the exact opposite of what it should 😆

Comment (kt3k) [reply yaml/parse_test.ts:1]: `JSON.parse()` doesn't seem throwing with duplicated keys?

```
> JSON.parse('{"a":1,"a":2}')
{ a: 2 }
```

**Review by kt3k — APPROVED** [review]
LGTM

File: yaml/parse_test.ts
Comment (iuioiua) [reply yaml/parse_test.ts:1]: Ok - then my understanding of `JSON.parse()` was wrong 😆

**Pull request merged**